src/d3a/models/market/one_sided.py

```diff
@@ -59,7 +59,8 @@ def offer(self, price: float, energy: float, seller: str) -> Offer:
         if energy <= 0:
             raise InvalidOffer()
 
-        price = price * (1 + self.transfer_fee_ratio) + self.transfer_fee_const * energy
+        if price != 0:
```

Comment:
I recall discussing about it at some point, but what is the reason for that change ? Even if a device is selling energy for free, I would expect that the IAAs would still need to get their fees.

Reply:
Yeah, but who is paying them? Do we allow negative rates?

Reply:
Ah I see, the new price is going to be 0, because it is relative to the initial price. In that case, I would suggest to leave it here as it is, and not forward the offer at all if its price is 0.

Reply:
Sorry, can you please be more explicit? Shall this function not return an offer if the price is 0? Because offer() is not only used in the forward_offer method. 
I could add 
```
        if price <= 0:
            raise InvalidOffer()
```
What do you think?

Reply:
What I meant, is to add a separate check in the forward_offer, that will not forward the offer if the target market's percentage fee is greater than zero and the offer price is zero or less. 
Raising an exception is not the desired behavior IMO, because this would crash the simulation just because an offer could not be forwarded.

Reply:
Understood. However, as mentioned, this method is not only called in the forward_offer method. So, even a storage can post an (initial) offer with price=0. Shall I add tests for that before each call? Sorry if i missed something.